Tests/AdyenTests/Adyen Tests/Components/Gift Card/GiftCardComponentTests.swift

```diff
@@ -0,0 +1,414 @@
+//
+//  GiftCardComponentTests.swift
+//  AdyenUIKitTests
+//
+//  Created by Mohamed Eldoheiri on 4/22/21.
+//  Copyright © 2021 Adyen. All rights reserved.
+//
+
+import XCTest
+@testable import Adyen
+@testable import AdyenCard
+
+class GiftCardComponentTests: XCTestCase {
+
+    var partialPaymentDelegate: PartialPaymentDelegateMock!
+
+    var readyToSubmitPaymentComponentDelegate: ReadyToSubmitPaymentComponentDelegateMock!
+
+    var delegateMock: PaymentComponentDelegateMock!
+
+    var sut: GiftCardComponent!
+
+    var paymentMethod: GiftCardPaymentMethod!
+
+    var errorView: FormErrorItemView? {
+        sut.viewController.view.findView(with: "AdyenCard.GiftCardComponent.errorItem")
+    }
+
+    var numberItemView: FormTextInputItemView? {
+        sut.viewController.view.findView(with: "AdyenCard.GiftCardComponent.numberItem")
+    }
+
+    var securityCodeItemView: FormTextInputItemView? {
+        sut.viewController.view.findView(with: "AdyenCard.GiftCardComponent.securityCodeItem")
+    }
+
+    var payButtonItemViewButton: UIControl? {
+        sut.viewController.view.findView(with: "AdyenCard.GiftCardComponent.payButtonItem.button")
+    }
+
+    override func setUp() {
+        super.setUp()
+        paymentMethod = GiftCardPaymentMethod(type: "testType", name: "testName", brand: "testBrand")
+        sut = GiftCardComponent(paymentMethod: paymentMethod, clientKey: Dummy.dummyClientKey)
+        sut.payment = nil // Missing Payment object
+        delegateMock = PaymentComponentDelegateMock()
+        sut.delegate = delegateMock
+        partialPaymentDelegate = PartialPaymentDelegateMock()
+        sut.partialPaymentDelegate = partialPaymentDelegate
+        readyToSubmitPaymentComponentDelegate = ReadyToSubmitPaymentComponentDelegateMock()
+        sut.readyToSubmitComponentDelegate = readyToSubmitPaymentComponentDelegate
+    }
+
+    func testCheckBalanceFailure() throws {
+
+        let didFailExpectation = expectation(description: "Expect delegateMock.onDidFail to be called.")
+        didFailExpectation.assertForOverFulfill = true
+        delegateMock.onDidFail = { error, component in
+            XCTAssertTrue(component === self.sut)
+            XCTAssertNotNil(error as? Dummy)
+            didFailExpectation.fulfill()
+        }
+
+        let onCheckBalanceExpectation = expectation(description: "Expect partialPaymentDelegate.onCheckBalance to be called.")
```

Comment:
Why new line for every single line in this file? 😅

Reply:
easier to read I guess, its soft on the eyes :-)